core/controller/src/main/scala/whisk/core/controller/Controller.scala

```diff
@@ -180,14 +180,24 @@ object Controller {
         // second argument.  (TODO .. seems fragile)
         val instance = if (args.length > 0) args(1).toInt else 0
 
-        // initialize the runtimes manifest
-        if (config.isValid && ExecManifest.initialize(config)) {
-            val port = config.servicePort.toInt
-            BasicHttpService.startService(actorSystem, "controller", "0.0.0.0", port, new ServiceBuilder(config, instance, logger))
-        } else {
+        def abort() = {
             logger.error(this, "Bad configuration, cannot start.")
             actorSystem.terminate()
             Await.result(actorSystem.whenTerminated, 30.seconds)
         }
+
+        if (!config.isValid) {
+            abort()
+            return
+        }
+
```

Comment:
> Everything not 0 should be an error

That's right. IIRC the default status code is `0`; we have the option to be explicit, by providing anything else but `0` so that the java program doesn't give the impression it has terminated normally. I believe we could also return an error message so that we don't need to look inside the logs to see which property was not correct.